src/uwtools/config/formats/ini.py

```diff
@@ -63,6 +64,8 @@ def dump_dict(path: OptionalPath, cfg: dict) -> None:
         # _final_ section, resulting in an anomalous trailing newline. To avoid this, write first to
         # memory, then strip the trailing newline.
 
+        config_check_depths(cfg, target_format="ini", mode="dump")
```

Comment:
Formats are already defined in a data class. Consider using those here.

Reply:
I've tried pulling from the input with `get_file_type(str(path))`

Reply:
Removing this for the data class